File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_a3/organic_8.java
// Linked List Reversal - organic submission 8
// Class-based with manual pointer swaps count.

class Node {
    int data;
    Node next;

    public Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    Node head;

    public LinkedList(int[] items) {
        this.head = null;
        if (items != null) {
            for (int i : items) {
                this.append(i);
            }
        }
    }

    public void append(int data) {
        Node n = new Node(data);
        if (this.head == null) {
            this.head = n;
            return;
        }
        Node cur = this.head;
        while (cur.next != null) {
            cur = cur.next;
        }
        cur.next = n;
    }

    public void reverse() {
        Node prevNode = null;
        Node curNode = this.head;
        int swaps = 0;
        while (curNode != null) {
            Node nxtNode = curNode.next;
            curNode.next = prevNode;
            prevNode = curNode;
            curNode = nxtNode;
            swaps++;
        }
        this.head = prevNode;
        System.out.println("performed " + swaps + " pointer swaps");
    }

    public void printList() {
        Node cur = this.head;
        StringBuilder sb = new StringBuilder();
        while (cur != null) {
            sb.append(cur.data);
            if (cur.next != null) sb.append(" ");
            cur = cur.next;
        }
        System.out.println(sb.toString());
    }

    public static void main(String[] args) {
        LinkedList ll = new LinkedList(new int[]{11, 22, 33, 44, 55});
        ll.printList();
        ll.reverse();
        ll.printList();
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_a3/organic_8.py
# Linked List Reversal - organic submission 8
# Class-based with __repr__, and reversal done with a manual index-based
# while loop plus an explicit swap-count print for debugging flavor.

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

    def __repr__(self):
        return f"Node({self.data!r})"


class LinkedList:
    def __init__(self, items=None):
        self.head = None
        if items:
            for i in items:
                self.append(i)

    def append(self, data):
        n = Node(data)
        if not self.head:
            self.head = n
            return
        cur = self.head
        while cur.next:
            cur = cur.next
        cur.next = n

    def reverse(self):
        prev_node = None
        cur_node = self.head
        swaps = 0
        while cur_node:
            nxt_node = cur_node.next
            cur_node.next = prev_node
            prev_node = cur_node
            cur_node = nxt_node
            swaps += 1
        self.head = prev_node
        print(f"performed {swaps} pointer swaps")

    def as_list(self):
        result = []
        cur = self.head
        while cur:
            result.append(cur.data)
            cur = cur.next
        return result


if __name__ == "__main__":
    ll = LinkedList([11, 22, 33, 44, 55])
    print(ll.as_list())
    ll.reverse()
    print(ll.as_list())
